crates/turborepo-ui/src/tui/app.rs

```diff
@@ -832,30 +816,13 @@ fn update(
         }
         Event::TogglePinnedTask => {
             app.is_task_selection_pinned = !app.is_task_selection_pinned;
-            let _ = preferences::Preferences::update_preference(
-                repo_root,
-                preferences::PreferenceFields::PinnedTaskSelection,
-                serde_json::Value::Bool(app.is_task_selection_pinned),
-            );
-
-            // Clear the selected task because the user has expressed that they don't care
-            // which task they're looking at right now. If they do care, they will either
-            // pin again or scroll through the task list.
-            if !app.is_task_selection_pinned {
-                let _ = preferences::Preferences::update_preference(
-                    repo_root,
-                    preferences::PreferenceFields::ActiveTask,
-                    serde_json::Value::String("".to_string()),
-                );
-            }
+            app.persist_active_task().ok();
         }
         Event::ToggleSidebar => {
-            app.has_sidebar = !app.has_sidebar;
-            let _ = preferences::Preferences::update_preference(
-                repo_root,
-                preferences::PreferenceFields::IsTaskListVisible,
-                serde_json::Value::Bool(app.has_sidebar),
-            );
+            let new_value = !app.preferences.is_task_list_visible();
```

Comment:
TODO: we should move this to be a method on `App` instead of a branch in our `update` match. The former is far easier to write tests for.